parity-util-mem/src/impls.rs

```diff
@@ -36,31 +34,26 @@ malloc_size_of_is_0!(std::time::Duration);
 
 malloc_size_of_is_0!(U64, U128, U256, U512, H32, H64, H128, H160, H256, H264, H512, H520, Bloom);
 
-macro_rules! impl_elastic_array {
-	($name: ident, $dummy: ident, $size: expr) => {
-		impl<T> MallocSizeOf for $name<T>
+macro_rules! impl_smallvec {
+	($size: expr) => {
+		impl<T> MallocSizeOf for SmallVec<[T; $size]>
 		where
 			T: MallocSizeOf,
 		{
-			fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
-				self[..].size_of(ops)
+			fn size_of(&self, _: &mut MallocSizeOfOps) -> usize {
+				if self.spilled() {
+					self.capacity() * core::mem::size_of::<T>()
```

Comment:
@cheme is this the way this should be done? I'm not sure I understand the purpose of the `MallocSizeOfOps` type and I'm not sure if this is accurate at all. See the test below also.

Reply:
ops is having way of measuring heap (depending on allocator or also possibly avoiding measuring something twice (case of Rc), but it is mainly a technical detail when implementing here.
In both case (spilled or not spilled) we should apply size_of on all item (for instance if there is a box in one of the items).
So your implementation correspond to shallow size of (direcly allocated), and we should add to it the value of `self.iter().sum(|a| a.size_of(ops))`.